File: 1927-maximum-ascending-subarray-sum/1927-maximum-ascending-subarray-sum.c
int maxAscendingSum (int* nums, int numsSize) {
    int maxSum = 0, sum = nums [0];
    for (int i = 1; i < numsSize; i++) {
        if (nums [i] > nums [i - 1]) sum += nums [i];
        else {
            maxSum = fmax (maxSum, sum);
            sum = nums [i];
        }
    }
    return fmax (maxSum, sum);
}

File: 1927-maximum-ascending-subarray-sum/1927-maximum-ascending-subarray-sum.java
class Solution {
    public int maxAscendingSum(int[] nums) {
        final int n = nums.length;
        int maxSum = 0, sum = nums [0];
        for (int i = 1; i < n; i++) {
            if (nums [i] > nums [i - 1]) { 
                sum += nums [i]; 
            }
            else {
                maxSum = Math.max (maxSum, sum);
                sum = nums [i];
            }
        }
        return Math.max (maxSum, sum);
    }
}